packages/dds/tree/src/feature-libraries/editable-tree/editableTreeSequence.ts

```diff
@@ -0,0 +1,349 @@
+ * the sequence fields by means of the cursors.
+ */
+export class ProxyTargetSequence extends Array<ProxyTarget | ProxyTargetSequence> {
+    private readonly target: ProxyTarget;
+    
+    constructor(
+        public readonly context: ProxyContext,
+        children?: (ProxyTarget | ProxyTargetSequence)[],
+        fieldCursor?: ITreeSubscriptionCursor,
+    ) {
+        super(...children ?? []);
+        if (fieldCursor === undefined) {
+            const rootCursor = context.forest.allocateCursor();
+            context.forest.tryMoveCursorTo(context.forest.root(context.forest.rootField), rootCursor);
+            this.target = context.createTarget(rootCursor);
+            rootCursor.free();
+        } else {
+            this.target = context.createTarget(fieldCursor);
+        }
+        const privateProperties: PropertyKey[] = ["target", "context"];
+        for (const propertyKey of privateProperties) {
+            Object.defineProperty(this, propertyKey,
+                { enumerable: false, writable: false, configurable: false, value: Reflect.get(this, propertyKey) });
+        }
+    }
+
+    public getAnchor(): Anchor {
+        return this.target.getAnchor();
+    }
+
+    public free(): void {
+        this.target.free();
+    }
+
+    public prepareAnchorForEdit(): Anchor {
+        return this.target.prepareAnchorForEdit();
+    }
+
+    public get primaryKey(): LocalFieldKey | undefined {
+        return this.target.getPrimaryArrayKey();
+    }
+
+    public get length(): number {
+        const primaryKey = this.primaryKey;
+        if (primaryKey !== undefined) {
+            return this.cursor.length(primaryKey);
+        }
+        // TODO: should return a cursor-based length as well
+        return super.length;
+    }
+
+    get cursor(): ITreeSubscriptionCursor {
+        return this.target.cursor;
+    }
+
+    public getType(key?: string, nameOnly?: boolean): TreeSchemaIdentifier | TreeSchema | undefined {
+        let typeName: TreeSchemaIdentifier | undefined = this.cursor.type;
+        const primaryKey = this.primaryKey;
+        if (key !== undefined) {
+            if (primaryKey !== undefined) {
+                const result = this.cursor.down(primaryKey, 0);
+                if (result === TreeNavigationResult.Ok) {
+                    typeName = this.cursor.seek(Number(key)) === TreeNavigationResult.Ok ? this.cursor.type : undefined;
+                    this.cursor.up();
+                }
+            } else {
+                typeName = this.cursor.seek(Number(key)) === TreeNavigationResult.Ok ? this.cursor.type : undefined;
+            }
+        }
+        if (nameOnly) {
+            return typeName;
+        }
+        if (typeName) {
+            return this.context.forest.schema.lookupTreeSchema(typeName);
+        }
+        return undefined;
+    }
+
+    /**
+     * Sets value of the node if index exists.
+     */
+    public setValue(index: number, value: unknown): boolean {
```

Comment:
```suggestion
    public setValue(index: number, value: Value): boolean {
```

Reply:
It looks like the code hasn't changed. Was that intentional?

Reply:
Nope. Fixed.